kernel/kernel-api/src/main/java/io/delta/kernel/internal/skipping/StatsSchemaHelper.java

```diff
@@ -0,0 +1,304 @@
+
+    public StatsSchemaHelper(StructType dataSchema) {
+        this.dataSchema = dataSchema;
+        Map<Column, Tuple2<Column, DataType>> logicalToPhysicalColumnAndDataType =
+            getLogicalToPhysicalColumnAndDataType(dataSchema);
+        this.logicalToPhysicalColumn = logicalToPhysicalColumnAndDataType.entrySet().stream()
+            .collect(
+                Collectors.toMap(
+                    Map.Entry::getKey,
+                    e -> e.getValue()._1 // map to just the column
+                )
+            );
+        this.logicalToDataType =
+            logicalToPhysicalColumnAndDataType.entrySet().stream()
+                .collect(
+                    Collectors.toMap(
+                        Map.Entry::getKey,
+                        e -> e.getValue()._2 // map to just the data type
+                    )
+                );
+    }
+
+    /** Given a logical column returns corresponding the MIN column in the statistic schema */
+    public Column getMinColumn(Column column) {
+        checkArgument(isSkippingEligibleMinMaxColumn(column),
+            String.format("%s is not a valid min column for data schema %s", column, dataSchema));
+        return getStatsColumn(column, MIN);
+    }
+
+    /** Given a logical column returns corresponding the MAX column in the statistic schema */
+    public Column getMaxColumn(Column column) {
+        checkArgument(isSkippingEligibleMinMaxColumn(column),
+            String.format("%s is not a valid min column for data schema %s", column, dataSchema));
+        return getStatsColumn(column, MAX);
+    }
+
+    /**
+     * Returns true if the given column is skipping-eligible using min/max statistics. This means
+     * the column exists, is a leaf column, and is of a skipping-eligible data-type.
+     */
+    public boolean isSkippingEligibleMinMaxColumn(Column column) {
+        return logicalToDataType.containsKey(column) &&
+            isSkippingEligibleDataType(logicalToDataType.get(column)) &&
+            // TODO for now we block using min/max columns of timestamps. JSON serialization
+            //   truncates to milliseconds. To safely use timestamp min/max stats we need to add
+            //   a millisecond to max statistics which requires time addition expression
+            !(logicalToDataType.get(column) instanceof TimestampType);
+    }
+
+    /**
+     * Returns true if the given column is skipping-eligible using null count statistics. This means
+     * the column exists and is a leaf column as we only collect stats for leaf columns.
+     */
+    public boolean isSkippingEligibleNullCountColumn(Column column) {
+        return logicalToPhysicalColumn.containsKey(column);
+    }
+
+    //////////////////////////////////////////////////////////////////////////////////
+    // Private static fields and methods
+    //////////////////////////////////////////////////////////////////////////////////
+
+    private static String NUM_RECORDS = "numRecords";
```

Comment:
i'd give (comment) some context as to what these variables are used for. All I know is that they are a string. what do they represent semantically? it's not like whenever I want to log "numRecords" to the console, I should use the NUM_RECORDS constant instard, right? so, clearly, they have some other intended use or meaning

Reply:
Added a comment